src/dotnet/APIView/APIView/NamedTypeAPIV.cs

```diff
@@ -0,0 +1,115 @@
+﻿using Microsoft.CodeAnalysis;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Text;
+
+namespace APIView
+{
+    /// <summary>
+    /// Class representing a C# named type (class, interface, delegate, enum, or struct). 
+    /// A named type can have a name, type, enum underlying type, events, fields, implemented 
+    /// classes/interfaces, methods, properties, type parameters, and/or other named types.
+    /// 
+    /// NamedTypeAPIV is an immutable, thread-safe type.
+    /// </summary>
+    public class NamedTypeAPIV
+    {
+        public string Name { get; }
+        public string Type { get; }
```

Comment:
Prefer enum instead of string constants where the set of values is known beforehand.